tools/RAiDER/llreader.py

```diff
@@ -53,27 +55,69 @@ def projection(self):
         return self._proj
 
 
-    def add_buffer(self, buffer):
-        '''
-        Check whether an extra lat/lon buffer is needed for raytracing
-        '''
-        # if raytracing, add a 1-degree buffer all around
-        try:
-            ll_bounds = self._bounding_box.copy()
-        except AttributeError:
-            ll_bounds = list(self._bounding_box)
-        ll_bounds[0] = np.max([ll_bounds[0] - buffer, -90])
-        ll_bounds[1] = np.min([ll_bounds[1] + buffer,  90])
-        ll_bounds[2] = np.max([ll_bounds[2] - buffer,-180])
-        ll_bounds[3] = np.min([ll_bounds[3] + buffer, 180])
-        return ll_bounds
+    def add_buffer(self, buffer=0.5, digits=2):
+        """ 
+        Add a fixed buffer to the AOI. Ensures cube is slighly larger than requested area.
+
+        Args: 
+            buffer (float)  - decimal degrees to be added to the bounding box
+            digits (int)    - number of decimal digits to include in the output
+        
+        Returns:
+            None. Updates self._bounding_box
+        
+        Example: 
+        >>> from RAiDER.models.hrrr import HRRR
+        >>> from RAiDER.llreader import BoundingBox
+        >>> wm = HRRR()
+        >>> aoi = BoundingBox([37, 38, -92, -91])
+        >>> aoi.add_buffer(buffer = 1.5 * wm.getLLRes())
+        >>> aoi.bounds()
+         [36.93, 38.07, -92.07, -90.93]
+        """
+        S, N, W, E = self.bounds()
+
+        S, N = np.max([S-buffer, -90]),  np.min([N+buffer, 90])
+        W, E = W-buffer, E+buffer # will need to handle dateline crossings elsewhere
+
+        self._bounding_box = [np.round(a, digits) for a in (S, N, W, E)]
+        return
+
+
+    def calc_buffer_ray(self, direction, lookDir='right', incAngle=30, maxZ=80, digits=2):
+        """ 
+        Calculate the buffer for ray tracing. This only needs to be done in the east-west
+        direction due to satellite orbits, and only needs extended on the side closest to 
+        the sensor.
+
+        Args: 
+            lookDir (str)      - Sensor look direction, can be "right" or "left"
+            losAngle (float)   - Incidence angle in degrees
+            maxZ (float)       - maximum integration elevation in km
+        """
+        S, N, W, E = self.bounds()
+        
+        # use a small look angle to calculate near range
+        lat_max = np.max([np.abs(S), np.abs(N)])
+        near    = maxZ * np.tan(np.deg2rad(incAngle))
+        buffer  = near / (np.cos(np.deg2rad(lat_max)) * 100)
+
+        # buffer on the side nearest the sensor
+        if ((lookDir == 'right') and (direction == 'asc')) or ((lookDir == 'left') and (direction == 'desc')):
+            W = W - buffer
+        else:
+            E = E + buffer
```

Comment:
Same comment here about date time crossing as you had for the other buffer

Reply:
@dbekaert @bbuzz31 On this topic it may be good to do some creative thinking, e.g., leaving longitudes unwrapped for -180-180 and converting to 0-360 when needed in order to avoid having two sub-AOIs defined (one >-180 and one <180). Didn't go too far with that thinking here, thought it may need its own PR.

Reply:
I agree, this should be a separate PR